products/surveys/backend/max_tools.py

```diff
@@ -0,0 +1,247 @@
+"""
+MaxTool for AI-powered survey creation.
+"""
+
+from typing import Any
+from pydantic import BaseModel, Field
+from langchain_core.prompts import ChatPromptTemplate
+from langchain_openai import ChatOpenAI
+from posthog.exceptions_capture import capture_exception
+
+from ee.hogai.tool import MaxTool
+from posthog.models import Team
+from .prompts import SURVEY_CREATION_SYSTEM_PROMPT
+from .survey_schema import SurveyCreationOutput, DEFAULT_SURVEY_APPEARANCE
+
+
+class SurveyCreatorArgs(BaseModel):
+    instructions: str = Field(description="Natural language description of the survey to create")
+
+
+class SurveyCreatorTool(MaxTool):
+    name: str = "create_survey"
+    description: str = "Create and optionally launch a survey based on natural language instructions"
+    thinking_message: str = "Creating your survey"
+
+    root_system_prompt_template: str = """
+    You are helping create surveys for this PostHog team.
+
+    Current context:
+    - Total surveys: {{{total_surveys_count}}}
+    - Recent surveys: {{{existing_surveys}}}
+
+    When creating surveys, consider the existing surveys to avoid duplication and suggest complementary survey strategies.
+    """
+
+    args_schema: type[BaseModel] = SurveyCreatorArgs
+
+    def _run_impl(self, instructions: str) -> tuple[str, dict[str, Any]]:
+        """
+        Generate survey configuration from natural language instructions.
+        """
+        try:
+            user = None
+            user_id = self.context.get("user_id")
+            if not user_id:
+                return "❌ Failed to create survey: User id not present on the context", {
+                    "error": "user_id_not_present"
+                }
+
+            try:
+                from posthog.models import User
+
+                user = User.objects.get(uuid=user_id)
+            except User.DoesNotExist:
+                return "❌ Failed to create survey: Invalid user", {"error": "invalid_user"}
+
+            # Get team for context
+            team = Team.objects.get(id=self._team_id)
```

Comment:
Same as above, from that PR you should be able to call `self._team`

Reply:
done